### PR TITLE
[TF C API] Add TF_SetShape C API

### PR DESCRIPTION
**Pull request by aadwived**
C API to set shape of tensor is missing. This API is required by TF Plugin to modify shapes of existing tensors.

Signed-off-by: Aakar Dwivedi <Aakar.Dwivedi@amd.com>
Reviewed-by: Chandra Kumar Ramasamy <ChandraKumar.Ramasamy@amd.com>

### CODE REVIEW

**Review by rohan100jain — APPROVED** [review]
Please wait for @sagunb to approve.

**Comment by penpornk** [comment]
> Thanks for your pull request! It looks like this may be your first contribution to a Google open source project. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> View this [failed invocation](https://github.com/tensorflow/tensorflow/pull/57008/checks?check_run_id=7671816092) of the CLA check for more information.
> 
> For the most up to date status, view the checks section at the bottom of the pull request.

Forgot to reply to this automated comment. I had verified that @aadwived had signed CLA with us before we merged this PR. Adding `cla: yes` tag.

**Review by sagunb — COMMENTED** [review]
Thanks for the changes. LGTM.

**Pull request merged**
